pymc/tests/test_distributions_timeseries.py

```diff
@@ -11,21 +11,94 @@
-from pymc.distributions.timeseries import AR, AR1, GARCH11, EulerMaruyama
+from pymc.distributions.timeseries import (
+    AR,
+    AR1,
+    GARCH11,
+    EulerMaruyama,
+    GaussianRandomWalk,
+    gaussianrandomwalk,
+)
 from pymc.model import Model
 from pymc.sampling import sample, sample_posterior_predictive
+from pymc.tests import test_distributions as td
 from pymc.tests.helpers import select_by_precision
+from pymc.tests.test_distributions_random import BaseTestDistributionRandom
+
+
+class TestGaussianRandomWalk(BaseTestDistributionRandom):
+    # Override default size for test class
+    size = None
 
-# pytestmark = pytest.mark.usefixtures("seeded_test")
-pytestmark = pytest.mark.xfail(reason="Timeseries not refactored")
+    pymc_dist = pm.GaussianRandomWalk
+    pymc_dist_params = {"mu": 1.0, "sigma": 2, "init": pm.Constant.dist(0), "steps": 4}
+    expected_rv_op_params = {"mu": 1.0, "sigma": 2, "init": pm.Constant.dist(0), "steps": 4}
+
+    checks_to_run = [
+        "check_pymc_params_match_rv_op",
+        "check_rv_inferred_size",
+    ]
+
+    def check_rv_inferred_size(self):
+        steps = self.pymc_dist_params["steps"]
+        sizes_to_check = [
+            None,
+            (),
+            1,
+            (1,),
+        ]
+        sizes_expected = [(steps + 1,), (steps + 1,), (1, steps + 1), (1, steps + 1)]
+        # sizes_to_check = [None, (), 1, (1,), 5, (4, 5), (2, 4, 2)]
+        # sizes_expected = [(), (), (1,), (1,), (5,), (4, 5), (2, 4, 2)]
+
+        for size, expected in zip(sizes_to_check, sizes_expected):
+            pymc_rv = self.pymc_dist.dist(**self.pymc_dist_params, size=size)
+            expected_symbolic = tuple(pymc_rv.shape.eval())
+            assert expected_symbolic == expected
+
+    def check_not_implemented(self):
+        with pytest.raises(NotImplementedError):
+            self.pymc_rv.eval()
+
+    def test_grw_inference(self):
+        mu, sigma, steps = 2, 1, 10000
+        obs = np.concatenate([[0], np.random.normal(mu, sigma, size=steps)]).cumsum()
+
+        with pm.Model():
+            _mu = pm.Uniform("mu", -10, 10)
+            _sigma = pm.Uniform("sigma", 0, 10)
+            # Workaround for bug in `at.diff` when data is constant
+            obs_data = pm.MutableData("obs_data", obs)
+            grw = GaussianRandomWalk("grw", _mu, _sigma, steps=steps, observed=obs_data)
+
+            trace = pm.sample()
+
+        recovered_mu = trace.posterior["mu"].mean()
+        recovered_sigma = trace.posterior["sigma"].mean()
+        np.testing.assert_allclose([mu, sigma], [recovered_mu, recovered_sigma], atol=0.2)
+
+
+class TestGRWScipy(td.TestMatchesScipy):
+    def test_grw(self):
+        self.check_logp(
+            pm.GaussianRandomWalk,
+            td.Vector(td.R, 10),
+            {"mu": td.R, "sigma": td.Rplus, "steps": td.Nat},
+            lambda value, mu, sigma: stats.norm.logpdf(value, mu, sigma).cumsum().sum(),
```

Comment:
This expression is wrong:
```suggestion
            lambda value, mu, sigma: stats.norm.logpdf(np.concatenate([value[0]], np.diff(value)], axis=-1)), , mu, sigma).cumsum().sum(),
```
Probably better to use a function than a lambda, to break that large one-liner